lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -925,52 +1003,51 @@ public function andHaving($predicate, ...$predicates) : self
      */
     public function orHaving($predicate, ...$predicates) : self
     {
-        return $this->appendPredicates('having', CompositeExpression::TYPE_OR, $predicate, ...$predicates);
+        $this->having = $this->appendToPredicate($this->having, CompositeExpression::TYPE_OR, $predicate, ...$predicates);
+
+        $this->state = self::STATE_DIRTY;
+
+        return $this;
     }
 
     /**
-     * Sets one or more predicates combined by the AND logic as the given query clause.
-     * Replaces any previously specified predicates.
+     * Creates a CompositeExpression from one or more predicates combined by the AND logic.
      *
+     * @param string|CompositeExpression $predicate
      * @param string|CompositeExpression ...$predicates
      *
-     * @return $this This QueryBuilder instance.
+     * @return string|CompositeExpression
      */
-    private function setPredicates(string $clause, ...$predicates) : self
+    private function createPredicate($predicate, ...$predicates)
     {
-        if (count($predicates) > 1) {
-            $predicate = new CompositeExpression(
-                CompositeExpression::TYPE_AND,
-                $predicates
-            );
-        } else {
-            $predicate = array_shift($predicates);
+        if (! $predicates) {
```

Comment:
Please use `if (count($predicates) < 1)` to treat `$predicates` as a generic collection and not depend on it being an array (same as above).

Reply:
Done.